### ISSUE TITLE
Improve shouldFail() similar to throws()?

### ISSUE THREAD

**Issue by unional** [spion/blue-tape#23]
That supports `shouldFail(p: Promise, expected: RegExp | Function, msg?: string)`

:rose:

**Comment by timoxley** [comment]
Agreed, every other tape assertion allows you to supply a message as last argument.

My workaround:

```js
function shouldFail (t, promise, message) {
  promise.then(
    () =>  t.fail(message),
    () => t.pass(message)
  )
}

// usage:
shouldFail(t, doBadThing(), 'bad thing should happen')
```

Ideally, `shouldFail` would piggy-back on top of the `Test.prototype.throws` logic so you don't need to reimplement all the argument & error message handling.

**Comment by tuckerconnelly** [comment]
Submitted a PR for this. Looks like this in practice:

```js
test('my component', assert => {
  assert.fails(functionReturningPromise(), 'this promise should fail')
})
```

**Comment by spion** [comment]
Its unclear whether "message" refers to the resulting message or whether the error message should be equal to the supplied message. Not merging this until there is good idea on how to avoid the confusion.

**Comment by spion** [comment]
Oops, looks I missed the original suggestion. Behaviour that is identical to `throws` would indeed be the least confusing.

**Comment by spion** [comment]
Released in 1.0.0 - there is a slight backward incompatibility so I bumped the major version.
